diff --git a/arch/arm/boot/dts/am335x-boneblack.dts b/arch/arm/boot/dts/am335x-boneblack.dts
index 6b71ad9..0299c79 100644
--- a/arch/arm/boot/dts/am335x-boneblack.dts
+++ b/arch/arm/boot/dts/am335x-boneblack.dts
@@ -60,6 +60,17 @@
 			0x1b0 0x03      /* xdma_event_intr0, OMAP_MUX_MODE3 | AM33XX_PIN_OUTPUT */
 		>;
 	};
+
+	gps_pps_pins: gps_pps_pins {
+		pinctrl-single,pins = <0x90 (PIN_INPUT_PULLDOWN | MUX_MODE7)>;
+	};
+	
+	uart1_pins: pinmux_uart1_pins {
+		pinctrl-single,pins = <
+		0x180 (PIN_INPUT_PULLUP | MUX_MODE0)	/* uart1_rxd.uart1_rxd */		
+		0x184 (PIN_OUTPUT_PULLDOWN | MUX_MODE0)	/* uart1_txd.uart1_txd */
+		>;
+	};	
 };
 
 &lcdc {
@@ -75,4 +86,19 @@
 		pinctrl-1 = <&nxp_hdmi_bonelt_off_pins>;
 		status = "okay";
 	};
+
+	pps {
+		compatible = "pps-gpio";
+		status = "okay";
+		pinctrl-names = "default";
+		pinctrl-0 = <&gps_pps_pins>;
+		gpios = <&gpio2 2 0>;
+		assert-rising-edge;
+	};
 };
+
+&uart1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart1_pins>;
+	status = "okay";
+};
\ No newline at end of file
