diff -Nru europa-2.6-6693/src/PLASMA/System/base/Jamfile europa-2.6-6693/src/PLASMA/System/base/Jamfile.old
--- europa-2.6-6693/src/PLASMA/System/base/Jamfile	2013-06-09 18:14:15.956332621 +0100
+++ europa-2.6-6693/src/PLASMA/System/base/Jamfile.old	2013-06-09 18:15:36.907863466 +0100
@@ -17,8 +17,8 @@
 	PSEngineImpl.cc
 	;
 
-SwigJava PSEngine.i : psengine : [ FDirName $(PLASMA_HOME) src Java PSEngine generated psengine ] : cpp : TinyXml Utils ConstraintEngine PlanDatabase RulesEngine NDDL TemporalNetwork Solvers System : PSEngine ;
-
-ModuleBaseSwig System : [ FSwigWrapperName PSEngine.i : cpp ] ;
+#SwigJava PSEngine.i : psengine : [ FDirName $(PLASMA_HOME) src Java PSEngine generated psengine ] : cpp : TinyXml Utils ConstraintEngine PlanDatabase RulesEngine NDDL TemporalNetwork Solvers System : PSEngine ;
+#
+#ModuleBaseSwig System : [ FSwigWrapperName PSEngine.i : cpp ] ;
 
 } # PLASMA_READY

