diff -Nru libswiftnav-0.11/clapack-3.2.1-CMAKE/F2CLIBS/libf2c/arith_cortex-a8.h libswiftnav-0.11.kk/clapack-3.2.1-CMAKE/F2CLIBS/libf2c/arith_cortex-a8.h
--- libswiftnav-0.11/clapack-3.2.1-CMAKE/F2CLIBS/libf2c/arith_cortex-a8.h	1970-01-01 01:00:00.000000000 +0100
+++ libswiftnav-0.11.kk/clapack-3.2.1-CMAKE/F2CLIBS/libf2c/arith_cortex-a8.h	2014-10-23 08:28:34.888000000 +0200
@@ -0,0 +1,7 @@
+/* arith.h definitions for ARM Cortex-A8
+ * Calculated by running arithchk on a Beaglebone Black. */
+#define IEEE_8087
+#define Arith_Kind_ASL 1
+#define Double_Align
+#define QNaN0 0x0
+#define QNaN1 0x7ff80000
