--- Python-2.7.2_host/Modules/Setup.dist	2011-06-11 17:46:26.000000000 +0200
+++ Python-2.7.2_target/Modules/Setup.dist	2013-05-21 17:52:37.161862158 +0200
@@ -460,7 +460,10 @@
 # Andrew Kuchling's zlib module.
 # This require zlib 1.1.3 (or later).
 # See http://www.gzip.org/zlib/
-#zlib zlibmodule.c -I$(prefix)/include -L$(exec_prefix)/lib -lz
+zlib zlibmodule.c -I$(prefix)/include -L$(exec_prefix)/lib -lz
+
+# Enabling the libbz2 in python:
+bz2 bz2module.c -I$(prefix)/include -L$(exec_prefix)/lib -lbz2
 
 # Interface to the Expat XML parser
 #
