--- linux-3.3.1/arch/arm/mach-omap2/Makefile	2012-04-02 18:32:52.000000000 +0100
+++ linux-3.3.1.rasm/arch/arm/mach-omap2/Makefile	2012-05-07 18:24:37.490190256 +0100
@@ -224,7 +224,7 @@
 					   board-zoom-display.o
 obj-$(CONFIG_MACH_CM_T35)		+= board-cm-t35.o
 obj-$(CONFIG_MACH_CM_T3517)		+= board-cm-t3517.o
-obj-$(CONFIG_MACH_IGEP0020)		+= board-igep0020.o
+obj-$(CONFIG_MACH_IGEP0020)		+= board-igep0020.o manta-axxx.o
 obj-$(CONFIG_MACH_OMAP3_TOUCHBOOK)	+= board-omap3touchbook.o
 obj-$(CONFIG_MACH_OMAP_4430SDP)		+= board-4430sdp.o
 obj-$(CONFIG_MACH_OMAP4_PANDA)		+= board-omap4panda.o
