--- linux-3.2.0-psp04.06.00.10/arch/arm/mach-omap2/board-am335xevm.c	2013-04-05 05:43:27.000000000 +0100
+++ linux-3.2.0-psp04.06.00.10.rasm/arch/arm/mach-omap2/board-am335xevm.c	2013-05-29 17:34:03.003915746 +0100
@@ -2362,6 +2362,9 @@
 			setup_beaglebone_old();
 		else
 			setup_beaglebone();
+	} else if (!strncmp("A335BNLT", config.name, 8)) {
+		daughter_brd_detected = false;
+                setup_beaglebone();
 	} else if (!strncmp("A335X_SK", config.name, 8)) {
 		daughter_brd_detected = false;
 		setup_starterkit();
