diff -Nru eglibc-2.15-16611/libc/nis/Makefile eglibc-2.15-16611-rasm/libc/nis/Makefile
--- eglibc-2.15-16611/libc/nis/Makefile	2012-01-09 21:16:30.000000000 +0000
+++ eglibc-2.15-16611-rasm/libc/nis/Makefile	2012-01-10 23:40:26.409018009 +0000
@@ -25,9 +25,9 @@
 
 aux			:= nis_hash
 
+headers			:= $(wildcard rpcsvc/*.[hx])
 distribute		:= nss-nis.h nss-nisplus.h nis_intern.h Banner \
-			   nisplus-parser.h nis_xdr.h nss \
-			   $(wildcard rpcsvc/*.[hx])
+			   nisplus-parser.h nis_xdr.h nss 
 
 # These are the databases available for the nis (and perhaps later nisplus)
 # service.  This must be a superset of the services in nss.
