diff -Nru perl-5.10.0.ori/utils/Makefile.SH perl-5.10.0/utils/Makefile.SH
--- perl-5.10.0.ori/utils/Makefile.SH	2007-12-18 10:47:08.000000000 +0000
+++ perl-5.10.0/utils/Makefile.SH	2011-02-26 16:30:24.000000000 +0000
@@ -28,7 +28,7 @@
 rm -f Makefile
 cat >Makefile <<!GROK!THIS!
 
-PERL = ../miniperl
+PERL = ../miniperl-cross
 REALPERL = ../perl
 RUN = $run  # Used mainly cross-compilation setups.
 
