diff -Nru nfs-utils-1.3.2/utils/statd/statd.c nfs-utils-1.3.2.rasm/utils/statd/statd.c
--- nfs-utils-1.3.2/utils/statd/statd.c	2015-01-30 20:42:15.000000000 +0000
+++ nfs-utils-1.3.2.rasm/utils/statd/statd.c	2015-09-09 16:06:03.761780966 +0100
@@ -393,7 +393,7 @@
 		simulator (--argc, ++argv);	/* simulator() does exit() */
 #endif
 
-	daemon_init(!(run_mode & MODE_NODAEMON));
+	daemon_init((run_mode & MODE_NODAEMON));
 
 	if (run_mode & MODE_LOG_STDERR) {
 		xlog_syslog(0);
