diff --git linux-4.4.66/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 30bbc37..7241be6 100644
--- linux-4.4.66/arch/arm/boot/dts/Makefile
+++ linux-4.4.66/arch/arm/boot/dts/Makefile
@@ -421,6 +421,7 @@
	omap3-igep0020-rev-f.dtb \
	omap3-igep0030.dtb \
	omap3-igep0030-rev-g.dtb \
+	omap3-lctr-a9xx.dtb \
	omap3-ldp.dtb \
	omap3-lilly-dbb056.dtb \
	omap3-n900.dtb \
diff -Nru linux-4.4.66/arch/arm/boot/dts/omap3-lctr-a9xx.dts linux-4.4.66.rasm/arch/arm/boot/dts/omap3-lctr-a9xx.dts
--- linux-4.4.66/arch/arm/boot/dts/omap3-lctr-a9xx.dts
+++ linux-4.4.66.rasm/arch/arm/boot/dts/omap3-lctr-a9xx.dts
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2014 Universidade do Porto - Faculdade de Engenharia
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include <dt-bindings/input/input.h>
+#include "omap3-igep0020.dts"
+
+/ {
+	leds {
+		boot {
+			linux,default-trigger = "heartbeat";
+		};
+	};
+
+	gpio_keys {
+		pinctrl-names = "default";
+		pinctrl-0 = <&gpio_keys_pins>;
+		compatible = "gpio-keys";
+
+		button@0 {
+			label = "Button 0";
+			gpios = <&gpio5 11 GPIO_ACTIVE_HIGH>;	/* gpio_139 */
+			linux,code = <BTN_0>;
+			debounce-interval = <5>;
+		};
+
+		button@1 {
+			label = "Button 1";
+			gpios = <&gpio5 10 GPIO_ACTIVE_HIGH>;	/* gpio_138 */
+			linux,code = <BTN_1>;
+			debounce-interval = <5>;
+		};
+
+		button@2 {
+			label = "Button 2";
+			gpios = <&gpio5 9 GPIO_ACTIVE_HIGH>;	/* gpio_137 */
+			linux,code = <BTN_2>;
+			debounce-interval = <5>;
+		};
+	};
+
+	pps {
+		compatible = "pps-gpio";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pps_pins>;
+		gpios = <&gpio5 8 GPIO_ACTIVE_HIGH>;
+		assert-rising-edge;
+	};
+};
+
+&omap3_pmx_core {
+	gpio_keys_pins: pinmux_gpio_keys_pins {
+		pinctrl-single,pins = <
+			0x136 (PIN_INPUT_PULLDOWN | MUX_MODE4)	/* gpio_137 - button 2 */
+			0x138 (PIN_INPUT_PULLDOWN | MUX_MODE4)	/* gpio_138 - button 1 */
+			0x13a (PIN_INPUT_PULLDOWN | MUX_MODE4)	/* gpio_139 - button 0 */
+		>;
+	};
+
+	pps_pins: pinmux_pps_pins {
+		pinctrl-single,pins = <
+			0x134 (PIN_INPUT_PULLDOWN | MUX_MODE4)	/* gpio_136 */
+		>;
+	};
+};
+
+&i2c2 {
+	rtc@68 {
+		compatible = "dallas,ds1307";
+		reg = <0x68>;
+	};
+};
+
+/* Disable WiFi/Bluetooth combo. */
+&mmc2 {
+	status = "disabled";
+};
+
+&vdd33 {
+	status = "disabled";
+};
+
+&lbee1usjyc_pdn {
+	status = "disabled";
+};
