diff -Nru linux-3.5.4/drivers/tty/serial/8250/8250.c linux-3.5.4.rasm/drivers/tty/serial/8250/8250.c
--- linux-3.5.4/drivers/tty/serial/8250/8250.c	2012-09-14 23:28:08.000000000 +0100
+++ linux-3.5.4.rasm/drivers/tty/serial/8250/8250.c	2015-03-19 14:49:54.026244446 +0000
@@ -207,8 +207,7 @@
 		.name		= "ST16654",
 		.fifo_size	= 64,
 		.tx_loadsz	= 32,
-		.fcr		= UART_FCR_ENABLE_FIFO | UART_FCR_R_TRIG_01 |
-				  UART_FCR_T_TRIG_10,
+		.fcr		= UART_FCR_ENABLE_FIFO | UART_FCR_R_TRIG_01,
 		.flags		= UART_CAP_FIFO | UART_CAP_EFR | UART_CAP_SLEEP,
 	},
 	[PORT_16850] = {
