--- a/Makefile.in	2007-03-10 23:18:00.000000000 +0000
+++ b/Makefile.in	2011-08-31 05:11:43.957621032 +0100
@@ -44,6 +44,7 @@
 TCLSH	= @V_TCLSH@
 TCL2C	= ./tcl2c++
 AR	= ar rc $(BLANK)
+DESTDIR = @prefix@
 
 RANLIB	= @RANLIB@
 INSTALL	= @INSTALL@
@@ -186,10 +186,10 @@
 	$(TCL2C) et_console $(CONSOLE_FILES) > embedded-console.cc
 
 install: force
-	$(INSTALL) -m 644 *.h @prefix@/include
-	$(INSTALL) tcl2c++ @prefix@/bin
-	$(INSTALL) $(LIB_TCLCL) @prefix@/lib
-	$(RANLIB)  @prefix@/lib/$(LIB_TCLCL)
+	$(INSTALL) -m 644 *.h $(DESTDIR)/include
+	$(INSTALL) tcl2c++ $(DESTDIR)/bin
+	$(INSTALL) $(LIB_TCLCL) $(DESTDIR)/lib
+	$(RANLIB)  $(DESTDIR)/lib/$(LIB_TCLCL)
 
 # libTcl is no longer installed as of 13-Nov-99
 #	-test -f @prefix@/lib/libTcl.a && rm -f @prefix@/lib/libTcl.a || true
