diff -Nru perl-5.10.0.ori/Makefile.SH perl-5.10.0/Makefile.SH
--- perl-5.10.0.ori/Makefile.SH	2011-02-26 16:03:47.000000000 +0000
+++ perl-5.10.0/Makefile.SH	2011-02-26 16:14:31.000000000 +0000
@@ -481,8 +481,8 @@
 uudmap.h: generate_uudmap\$(EXE_EXT)
 	\$(RUN) ./generate_uudmap\$(EXE_EXT) >uudmap.h
 
-generate_uudmap\$(EXE_EXT): generate_uudmap\$(OBJ_EXT)
-	\$(CC) -o generate_uudmap \$(LDFLAGS) generate_uudmap\$(OBJ_EXT) \$(libs)
+generate_uudmap\$(EXE_EXT): 
+	\$(HOSTCC) -o generate_uudmap \$(LDFLAGS) generate_uudmap.c \$(libs)
 
 !GROK!THIS!
 $spitshell >>Makefile <<'!NO!SUBS!'
@@ -919,11 +919,11 @@
 INSTALL_DEPENDENCE = all
 
 install.perl:	$(INSTALL_DEPENDENCE) installperl
-	$(LDLIBPTH) $(RUN) ./perl installperl --destdir=$(DESTDIR) $(INSTALLFLAGS) $(STRIPFLAGS)
+	$(LDLIBPTH) $(RUN) ./miniperl-cross installperl --destdir=$(DESTDIR) $(INSTALLFLAGS) $(STRIPFLAGS)
 	-@test ! -s extras.lst || $(MAKE) extras.install
 
 install.man:	all installman
-	$(LDLIBPTH) $(RUN) ./perl installman --destdir=$(DESTDIR) $(INSTALLFLAGS)
+	$(LDLIBPTH) $(RUN) echo installman --destdir=$(DESTDIR) $(INSTALLFLAGS)
 
 # XXX Experimental. Hardwired values, but useful for testing.
 # Eventually Configure could ask for some of these values.
