--- gcc-4.6.3/libjava/configure.host	2010-12-06 00:50:04.000000000 +0000
+++ gcc-4.6.3.rasm/libjava/configure.host	2012-10-30 20:29:06.192935517 +0000
@@ -115,7 +115,7 @@
   mips*-*)
 	libgcj_interpreter=yes
 	;;
-  i686-*|i586-*|i486-*|i386-*)
+  i786-*|i686-*|i586-*|i486-*|i386-*)
 	sysdeps_dir=i386
 	# With -fomit-frame-pointer -maccumulate-outgoing-args (implied),
 	# the .text section of libgcj.so is 30k larger, and the .eh_frame
