diff -uNr fconfig/fconfig.c fconfig.mod/fconfig.c
--- fconfig/fconfig.c	2010-02-05 05:01:14.000000000 +0000
+++ fconfig.mod/fconfig.c	2010-02-05 05:02:37.000000000 +0000
@@ -103,7 +103,7 @@
 struct config_data *get_fconfig_handle(struct config_data *data, 
 	uint8_t *dev, int offset, mode_t mode)
 {
-	uint16_t count;
+	uint32_t count;
 
 	if ((data->fd = open(dev, mode)) < 0) {
 		MESSAGE(VERB_LOW, "Failed to open device or file %s!\n", dev);
diff -uNr fconfig/ftypes.h fconfig.mod/ftypes.h
--- fconfig/ftypes.h	2010-02-05 05:01:14.000000000 +0000
+++ fconfig.mod/ftypes.h	2010-02-05 05:02:11.000000000 +0000
@@ -32,7 +32,7 @@
 #define MAX_STRING_LENGTH 128
 
 //CYGNUM_REDBOOT_FLASH_CONFIG_SIZE
-#define MAX_CONFIG_DATA 4096
+#define MAX_CONFIG_DATA 0x20000
 
 /*
  * RedBoot flash configuration type description. 
