--- apache-log4cxx-0.10.0/src/main/cpp/socketoutputstream.cpp	2008-04-01 00:34:09.000000000 +0200
+++ apache-log4cxx-0.10.0/src/main/cpp/socketoutputstream.cpp	2013-04-16 12:53:30.646181613 +0200
@@ -15,6 +15,7 @@
  * limitations under the License.
  */
 
+#include <cstring>
 #include <log4cxx/logstring.h>
 #include <log4cxx/helpers/socketoutputstream.h>
 #include <log4cxx/helpers/socket.h>
@@ -49,7 +50,7 @@ void SocketOutputStream::write(ByteBuffe
   if (buf.remaining() > 0) {
     size_t sz = array.size();
     array.resize(sz + buf.remaining());
-    memcpy(&array[sz], buf.current(), buf.remaining());
+    std::memcpy(&array[sz], buf.current(), buf.remaining());
     buf.position(buf.limit());
   }
 }
