diff -Nru gcc-4.6.3/gcc/config/i386/t-linux64 gcc-4.6.3.rasm/gcc/config/i386/t-linux64
--- gcc-4.6.3/gcc/config/i386/t-linux64	2009-04-21 20:03:23.000000000 +0100
+++ gcc-4.6.3.rasm/gcc/config/i386/t-linux64	2012-12-11 19:01:42.549964639 +0000
@@ -24,8 +24,8 @@
 # MULTILIB_OSDIRNAMES according to what is found on the target.
 
 MULTILIB_OPTIONS = m64/m32
-MULTILIB_DIRNAMES = 64 32 
-MULTILIB_OSDIRNAMES = ../lib64 $(if $(wildcard $(shell echo $(SYSTEM_HEADER_DIR))/../../usr/lib32),../lib32,../lib)
+MULTILIB_DIRNAMES = 64 32
+MULTILIB_OSDIRNAMES = ../lib $(if $(wildcard $(shell echo $(SYSTEM_HEADER_DIR))/../../usr/lib32),../lib32,../lib)
 
 LIBGCC = stmp-multilib
 INSTALL_LIBGCC = install-multilib
